src/couch_changes.erl

```diff
@@ -336,6 +343,16 @@ get_doc_ids(_) ->
     throw({bad_request, no_doc_ids_provided}).
 
 
+get_selector({json_req, {Props}}) ->
+    check_selector(couch_util:get_value(<<"selector">>, Props));
+get_selector(#httpd{method='POST'}=Req) ->
+    couch_httpd:validate_ctype(Req, "application/json"),
+    {Props} = couch_httpd:json_body_obj(Req),
+    check_selector(couch_util:get_value(<<"selector">>, Props));
```

Comment:
Seems like you repeat yourself here. How about to change that branch into:
```
get_selector(#httpd{method='POST'}=Req) ->
   couch_httpd:validate_ctype(Req, "application/json"),
   get_selector({json_req, couch_httpd:json_body_obj(Req)});
```

You'll also produce bad request instead of server internal error with function clause in case of invalid request json data, like array.